sympy/solvers/solveset.py

```diff
@@ -847,6 +847,9 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if not free_symbols:
         b = Eq(f, 0)
+        if isinstance(symbol, Number):
```

Comment:
I think, instead of checking specifically `Number` you must check whether it is a symbol or not. means this :  `getattr(symbol, 'is_Symbol', False)` .it can handle all other kind of non-symbols. Is there any problem in doing this ?